### DISCUSSION TITLE
Can't implement Sign In function

### DISCUSSION THREAD

**Discussion by mihaimazareanu** [neuefische/questions#369] [Web] [answered]
Hi,

It seems that whatever I do, I just can't get to make the `sign in` button work. 

I have created 3 state variables and a useEffect to fetch the data:

    const [users, setUsers] = useState([]);
    const [loginFilter, setLoginFilter] = useState({
    firstName: "",
    lastName: "",
    password: "",
    });
    const [signedin, setSignedin] = useState(false);

    useEffect(() => {
    const getUsers = async () => {
      try {
        const url = loginFilter
          ? `/api/users?firstName=${loginFilter.firstName}`
          : `/api/users`;
        const response = await fetch(url);
        if (response.ok) {
          const data = await response.json();
          setUsers(data);
        } else {
          throw new Error(`Fetch failed with Status: ${response.status}`);
        }
      } catch (error) {
        console.log(error);
        alert(error.message);
      }
    };
    getUsers();
    }, [loginFilter]);

Then I created the `handleSubmit` function:

    const handleSubmitSigninForm = event => {
    event.preventDefault();
    users.map(user => {
      user.firstName === loginFilter.firstName
        ? setSignedin(true)
        : alert(`User not found`);
    });
    };

However, nothing visible happens when the button is pressed.

Here is the PR with my code, I would appreciate it if someone could help me before I go crazy... 🙈 

https://github.com/mihaimazareanu/capstone-project-kidsfi/pull/20

**Comment by mihaimazareanu** [comment, marked as answer]
Nevermind, I had a mistake in my code in `/api/users.js` 🙈

**Reply by actyvystom** [reply]
That is strange - imho your code from above shouldn't call `/api/users.js`...
because
your `useEffect` will run also with default setting of your `loginFilter` state variable as you are checking the whole object in your ternary statement: 
```js
const url = loginFilter
      ? `/api/users?firstName=${loginFilter.firstName}`
      : `/api/users`;
```
So, your fetch will run with an empty firstName string and set your user state variable with an empty user...
